src/main/java/net/snowflake/ingest/streaming/internal/ExternalVolumeManager.java

```diff
@@ -0,0 +1,172 @@
+/*
+ * Copyright (c) 2024 Snowflake Computing Inc. All rights reserved.
+ */
+
+package net.snowflake.ingest.streaming.internal;
+
+import java.io.IOException;
+import java.util.Calendar;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+import net.snowflake.client.jdbc.SnowflakeSQLException;
+import net.snowflake.ingest.connection.IngestResponseException;
+import net.snowflake.ingest.utils.ErrorCode;
+import net.snowflake.ingest.utils.SFException;
+
+class ExternalVolumeLocation {
+  public final String dbName;
+  public final String schemaName;
+  public final String tableName;
+
+  public ExternalVolumeLocation(String dbName, String schemaName, String tableName) {
+    this.dbName = dbName;
+    this.schemaName = schemaName;
+    this.tableName = tableName;
+  }
+}
+
+/** Class to manage multiple external volumes */
+class ExternalVolumeManager<T> implements StorageManager<T, ExternalVolumeLocation> {
+  // Reference to the external volume per table
+  private final Map<String, StreamingIngestStorage<T, ExternalVolumeLocation>> externalVolumeMap;
+
+  // name of the owning client
+  private final String clientName;
+
+  // role of the owning client
+  private final String role;
+
+  // Reference to the Snowflake service client used for configure calls
+  private final SnowflakeServiceClient snowflakeServiceClient;
+
+  // Client prefix generated by the Snowflake server
+  private final String clientPrefix;
+
+  /**
+   * Constructor for ExternalVolumeManager
+   *
+   * @param isTestMode whether the manager in test mode
+   * @param role the role of the client
+   * @param clientName the name of the client
+   * @param snowflakeServiceClient the Snowflake service client used for configure calls
+   */
+  ExternalVolumeManager(
+      boolean isTestMode,
+      String role,
+      String clientName,
+      SnowflakeServiceClient snowflakeServiceClient) {
+    this.role = role;
+    this.clientName = clientName;
+    this.snowflakeServiceClient = snowflakeServiceClient;
+    this.externalVolumeMap = new ConcurrentHashMap<>();
+    try {
+      this.clientPrefix =
+          isTestMode
+              ? "testPrefix"
+              : this.snowflakeServiceClient
+                  .clientConfigure(new ClientConfigureRequest(role))
+                  .getClientPrefix();
+    } catch (IngestResponseException | IOException e) {
+      throw new SFException(e, ErrorCode.CLIENT_CONFIGURE_FAILURE, e.getMessage());
+    }
+  }
+
+  /**
+   * Given a channel context, return the target storage by looking up the table name
+   *
+   * @param channelFlushContext the channel flush context containing the table name
+   * @return target storage
+   */
+  @Override
+  public StreamingIngestStorage<T, ExternalVolumeLocation> getStorage(
+      ChannelFlushContext channelFlushContext) {
```

Comment:
could we only pass in the table name instead? Looks like that only what we needed